### PR TITLE
[WIP] feat:notion connector

### PR DESCRIPTION
**Pull request by shrikantNemiwal1**
## Description
Raised new pr for these changes : https://github.com/pipeshub-ai/pipeshub-ai/pull/1302
Added Notion connector with OAuth 2.0 authentication. Implemented user synchronization that fetches all users from Notion workspace, filters for person users (excludes bots), and retrieves email addresses for each user. 

Modified the generic OAuth service to support Notion's non-standard token exchange format (Basic Auth header + JSON body instead of form-encoded). Added Notion to Node.js connector type definitions across multiple files.

Created NotionApp class, registered connector in factory, and added example.py for standalone testing.

## Main Changes

- Notion connector with OAuth authentication and user sync
- OAuth service enhancement for Basic Auth + JSON body (Notion-specific)
- Notion type definitions in Node.js backend (ServiceType, ConnectorNames, APP_TYPES, ConnectorName)
- NotionApp class with proper enum handling
- Connector cleanup and resource management
- Standalone example script for testing

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

Manually

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/python/app/connectors/sources/notion/connector.py [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+                        if app_user:
+                            app_users.append(app_user)
+                        else:
+                            total_skipped += 1
+
+                    except Exception as e:
+                        self.logger.error(f"❌ Failed to process user {user_id}: {e}")
+                        total_skipped += 1
+                        continue
+
```
[the remark this answers is not in the document]
Comment (shrikantNemiwal1) [reply backend/python/app/connectors/sources/notion/connector.py]: fixed

File: backend/python/app/connectors/core/base/token_service/oauth_service.py [hunk 2]
```diff
@@ -161,54 +161,126 @@ async def exchange_code_for_token(self, code: str, state: Optional[str] = None,
...
+                token_data = await response.json()
+        else:
+            # Standard form-encoded body
+            async with session.post(self.config.token_url, data=data, headers=headers) as response:
+                if response.status == HttpStatusCode.FORBIDDEN.value:
+                    # Log additional details for 403 errors (common with expired/invalid refresh tokens)
+                    error_text = await response.text()
+                    raise Exception(f"Token refresh failed with 403 Forbidden. This usually means the refresh token has expired or is invalid. Response: {error_text}")
+                response.raise_for_status()
+                token_data = await response.json()
```
[the remark this answers is not in the document]
Comment (shrikantNemiwal1) [reply backend/python/app/connectors/core/base/token_service/oauth_service.py]: fixed

File: backend/python/app/connectors/core/base/token_service/oauth_service.py [hunk 3]
```diff
@@ -161,54 +161,126 @@ async def exchange_code_for_token(self, code: str, state: Optional[str] = None,
...
+                    masked_client_id = self.config.client_id[:FIRST_8_CHARS] + "..." if len(self.config.client_id) > FIRST_8_CHARS else "***"
+                    error_msg = (
+                        f"OAuth token exchange failed with status {response.status}. "
+                        f"Token URL: {self.config.token_url}, "
+                        f"Redirect URI: {self.config.redirect_uri}, "
+                        f"Client ID (masked): {masked_client_id}, "
+                        f"Response: {error_text}"
+                    )
+                    raise Exception(error_msg)
+                token_data = await response.json()
```
[the remark this answers is not in the document]
Comment (shrikantNemiwal1) [reply backend/python/app/connectors/core/base/token_service/oauth_service.py]: fixed
